### ISSUE TITLE
Binder: Multipart io.Reader parameters needs to be closed.

### ISSUE THREAD

**Issue by klauspost** [revel/revel#756]
I have tracked down an issue with revel, where temporary files are leaked.

I have a controller function like this:

```func (c File) Replace(id string, file io.Reader, metadata string) revel.Result {```

Short story is, if I upload a file larger than 32MB, I get this error:
```params.go:121: Error removing temporary files: remove c:\temp\win\multipart-265323653: The process cannot access the file, since it is in use by another process.```

The reason for this is, that the file is larger than 32MB, so "req.ParseMultipartForm" in "params.go" places the file on disk. Very reasonable.

However, this also means the "io.Reader" must be closed for the system to be able to delete the file after processing it. This is not done.

So whenever Revel binds a parameter, it should check if it implements io.Closer and call it once the controller has been called.

The workarounds are:
* Test if the supplied reader is io.Closer in the controller (this should be handled by Revel)
* Use "os.File" for the controller. Silly, since that would copy it to disk every time.

**Comment by pedromorgan** [comment]
Is this fixed ? assume it is ?

**Comment by anonx** [comment]
@pedromorgan No, it isn't fixed.

**Comment by brendensoares** [comment]
@pedromorgan this needs to be addressed in the code before we close.

**Comment by brendensoares** [comment]
@klauspost @pushrax @anonx how are we able to [delete the `tmpFile`](https://github.com/revel/revel/blob/develop/params.go#L117) before we invoke the controller action?  How would the action be able to access the contents of the `tmpFile`?

@klauspost What are the steps to reproduce this so that we can make sure it's properly addressed?

**Comment by klauspost** [comment]
I had a look at it, and it seems like this could be handled by adding a simple check in the "ActionInvoker" controller:

```Go
	for _, arg := range c.MethodType.Args {
		// If they accept a websocket connection, treat that arg specially.
		var boundArg reflect.Value
		if arg.Type == websocketType {
			boundArg = reflect.ValueOf(c.Request.Websocket)
		} else {
			boundArg = Bind(c.Params, arg.Name, arg.Type)
			// If the argument is a closer, defer a Close call, so we don't risk leaks.
			closer, ok := boundArg.Interface().(io.Closer)
			if ok {
				defer closer.Close()
			}
		}
		methodArgs = append(methodArgs, boundArg)
	}
```
This fixes the issue for me.

"bindByteArray" should just be closed in the binder after the file has been read.


To reproduce:
* Add POST controller, similar to this:
```func (c File) Add(file io.Reader) revel.Result {
  return c.RenderJson("Success")
}
``
* Send a file >32MB with a POST.
The temporary file will now remain visible on Linux, and the error above will show on Windows.

**Comment by brendensoares** [comment]
@klauspost we will be able to fix this when the new `binding` package(#872) is complete in v0.13.

**Comment by jeevatkm** [comment]
#872 might take time. @klauspost suggestion will work in this case, so I'm applying it.
